File: frontend/ItalianRestaurantFront/src/app/models/meal.ts
export class MealResponse{
  id: number
  name: string
  imgPath: string
  description: string
  price: number
  mealCategory: Category

  constructor(id: number, name: string, imgPath: string, description: string, price: number, mealCategory: Category) {
    this.id = id;
    this.name = name;
    this.imgPath = imgPath;
    this.description = description;
    this.price = price;
    this.mealCategory = mealCategory;
  }

  equals(other: MealResponse): boolean {
    return this.name === other.name &&
      this.imgPath === other.imgPath &&
      this.description === other.description &&
      this.price === other.price;
  }
}

export class Meal{
  id: number
  name: string
  imgPath: string
  description: string
  price: number
  mealCategory: string

  constructor(id: number,name: string, imgPath: string, description: string, price: number, mealCategory: string) {
    this.id = id
    this.name = name;
    this.imgPath = imgPath;
    this.description = description;
    this.price = price;
    this.mealCategory = mealCategory;
  }

  equals(other: MealResponse): boolean {
    return this.name === other.name &&
      this.imgPath === other.imgPath &&
      this.description === other.description &&
      this.price === other.price;
  }
}

export class Category{
  name: string;

  constructor(name: string) {
    this.name = name;
  }
}

File: src/main/java/com/example/italianrestaurant/meal/Meal.java
package com.example.italianrestaurant.meal;

import com.example.italianrestaurant.meal.mealcategory.MealCategory;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.Hibernate;

import java.util.Objects;

@Entity
@Table(name = "meals")
@Getter
@Setter
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Meal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private String name;
    private String imgPath;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(nullable = false)
    @ToString.Exclude
    private MealCategory mealCategory;
    private String description;
    @Column(nullable = false)
    private double price;


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Meal meal = (Meal) o;
        return getId() != null && Objects.equals(getId(), meal.getId());
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}
